app/src/main/java/org/wikipedia/page/leadimages/LeadImagesHandler.java

```diff
@@ -131,41 +132,52 @@ private void updateCallToAction() {
         }
         String imageTitle = "File:" + getPage().getPageProperties().getLeadImageName();
         disposables.add(Observable.zip(MediaHelper.INSTANCE.getImageCaptions(imageTitle),
-                ServiceFactory.get(getTitle().getWikiSite()).getImageInfo(imageTitle, WikipediaApp.getInstance().getAppOrSystemLanguageCode()), Pair::new)
+                ServiceFactory.get(getTitle().getWikiSite()).getImageInfo(imageTitle, WikipediaApp.getInstance().getAppOrSystemLanguageCode()),
+                ServiceFactory.get(new WikiSite(Service.COMMONS_URL)).getProtectionInfo(imageTitle), (captions, imageInfoRsp, protectionInfoRsp) -> {
+                    boolean protectedFile = false;
+                    for (Protection protection : protectionInfoRsp.query().firstPage().protection()) {
+                        if (protection.getType().equals("edit") && !protectionInfoRsp.query().userInfo().getGroups().contains(protection.getLevel())) {
+                            protectedFile = true;
+                            break;
+                        }
+                    }
+                    return protectedFile ? null : new Pair<>(captions, imageInfoRsp);
```

Comment:
You must never return `null` for the result of an `Observable`. This will cause an internal exception in RxJava, and will be passed on to onError, and will never make it into the `subscribe` block.

Reply:
You're right. Done!